server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyToResponseBody.scala

```diff
@@ -1,49 +1,80 @@
 package sttp.tapir.server.netty.internal
 
-import io.netty.buffer.{ByteBuf, Unpooled}
+import io.netty.buffer.Unpooled
+import io.netty.handler.codec.http.HttpChunkedInput
+import io.netty.handler.stream.ChunkedFile
 import sttp.capabilities
 import sttp.model.HasHeaders
 import sttp.tapir.capabilities.NoStreams
 import sttp.tapir.server.interpreter.ToResponseBody
-import sttp.tapir.{CodecFormat, RawBodyType, WebSocketBodyOutput}
+import sttp.tapir.server.netty.NettyResponse
+import sttp.tapir.{CodecFormat, FileRange, RawBodyType, WebSocketBodyOutput}
 
-import java.io.InputStream
+import java.io.{InputStream, RandomAccessFile}
 import java.nio.ByteBuffer
 import java.nio.charset.Charset
-import java.nio.file.Files
 
-class NettyToResponseBody extends ToResponseBody[ByteBuf, NoStreams] {
+class NettyToResponseBody extends ToResponseBody[NettyResponse, NoStreams] {
   override val streams: capabilities.Streams[NoStreams] = NoStreams
 
-  override def fromRawValue[R](v: R, headers: HasHeaders, format: CodecFormat, bodyType: RawBodyType[R]): ByteBuf = {
+  override def fromRawValue[R](v: R, headers: HasHeaders, format: CodecFormat, bodyType: RawBodyType[R]): NettyResponse = {
     bodyType match {
       case RawBodyType.StringBody(charset) =>
-        Unpooled.wrappedBuffer(v.toString.getBytes(charset))
+        val bytes = v.asInstanceOf[String].getBytes(charset)
+        Left(Unpooled.wrappedBuffer(bytes))
+
       case RawBodyType.ByteArrayBody =>
         val bytes = v.asInstanceOf[Array[Byte]]
-        Unpooled.wrappedBuffer(bytes)
+        Left(Unpooled.wrappedBuffer(bytes))
+
       case RawBodyType.ByteBufferBody =>
         val byteBuffer = v.asInstanceOf[ByteBuffer]
-        Unpooled.wrappedBuffer(byteBuffer)
+        Left(Unpooled.wrappedBuffer(byteBuffer.array()))
 
       case RawBodyType.InputStreamBody =>
         val stream = v.asInstanceOf[InputStream]
-        Unpooled.wrappedBuffer(stream.readAllBytes())
+        Left(Unpooled.wrappedBuffer(stream.readAllBytes()))
+
+      case RawBodyType.FileBody =>
+        val fileRange = v.asInstanceOf[FileRange]
+        Right(wrap(fileRange))
 
-      case RawBodyType.FileBody         => Unpooled.wrappedBuffer(Files.readAllBytes(v.file.toPath))
       case _: RawBodyType.MultipartBody => ???
     }
   }
 
+  private def wrap(content: FileRange): (HttpChunkedInput, Long) = {
+    val file = content.file
+    val maybeRange = for {
+      range <- content.range
+      start <- range.start
+      end <- range.end
+    } yield (start, end + NettyToResponseBody.IncludingLastOffset)
+
+    maybeRange match {
+      case Some((start, end)) => {
+        val randomAccessFile = new RandomAccessFile(file, NettyToResponseBody.ReadOnlyAccessMode)
+        (new HttpChunkedInput(new ChunkedFile(randomAccessFile, start, end - start, NettyToResponseBody.DefaultChunkSize)), end - start)
```

Comment:
does this always send content using `Transfer-Encoding: chunked`? That's usually only needed when the response length is unknown

Reply:
It never does that automatically. If needed, setting such a header needs to be done manually. The main reason for using `HttpChunkedInput` is because it will write the end marker (LastHttpContent) for us.

Reply:
So I checked the docs, in `HttpChunkedInput` it says that you should set `Transfer-Encoding: chunked`. Maybe we should make the response type more general, returning a `ChunkedInput`, which would allow us to return `ChunkedFile` directly?

Plus, shouldn't we set `Transfer-Encoding: chunked` in case the length is unknown? In other interpretrs we often return the content-length as part of the body representation, exactly for this reason: to set the transfer encoding & content length headers properly (see e.g. http4s)

Reply:
As far as I understand, you can use `HttpChunkedInput` and do NOT set `Transfer-Encoding: chunked` as shown in [netty's usage example](https://github.com/netty/netty/blob/d34212439068091bcec29a8fad4df82f0a82c638/example/src/main/java/io/netty/example/http/file/HttpStaticFileServerHandler.java#L205). Nonetheless, when we do not know the length on the response `Transfer-Encoding: chunked` should be set. Thanks!